lambdas/question/src/main/java/uk.gov.di.ipv.cri.kbv.api.handler/QuestionHandler.java

```diff
@@ -148,23 +154,23 @@ private void respondWithQuestionFromExperianThenStoreInDb(
             throws IOException, InterruptedException {
         // we should fall in this block once only
         // fetch a batch of questions from experian kbv wrapper
-        if (kbvItem.getAuthorizationCode() != null) {
+        if (kbvItem.getStatus() != null) {
             response.withStatusCode(HttpStatusCode.NO_CONTENT);
             return;
         }
         QuestionsResponse questionsResponse = this.kbvService.getQuestions(questionRequest);
-        if (questionsResponse.hasQuestions()) {
+        if (questionsResponse != null && questionsResponse.hasQuestions()) {
             questionState.setQAPairs(questionsResponse.getQuestions());
             Optional<Question> nextQuestion = questionState.getNextQuestion();
             response.withStatusCode(HttpStatusCode.OK);
             response.withBody(objectMapper.writeValueAsString(nextQuestion.get()));
 
             String state = objectMapper.writeValueAsString(questionState);
             kbvItem.setQuestionState(state);
-
             kbvItem.setAuthRefNo(questionsResponse.getControl().getAuthRefNo());
             kbvItem.setUrn(questionsResponse.getControl().getURN());
-            kbvStorageService.update(kbvItem);
+            kbvItem.setExpiryDate(Instant.now().getEpochSecond() + "");
```

Comment:
We want to set this to the session's expiry TTL (grab from configurationService), as this currently sets it to immediately expire